### DISCUSSION TITLE
DC input possible?

### DISCUSSION THREAD

**Discussion by ThisIsTheOnlyUsernameAvailable** [HASwitchPlate/HASPone#114] [Q&A] [answered]
Thanks very much for your work on this amazing project!

I'm uncomfortable with bringing mains AC directly into this enclosure.  Is it possible to use an external 5v source (eg a USB charger) to power it directly?  If so, which pads should be wired to the external source?

Many thanks!

**Comment by fvanroie** [comment, marked as answer]
There are DC-DC options available for several low voltage transformers, like 9-18VDC. Check out the [wiki page](https://github.com/aderusha/HASwitchPlate/blob/master/Documentation/03_Electronics_Assembly.md#dc-power-option) for details.

Providing 5V directly to the PCB shouldn't be a problem either.

**Reply by ThisIsTheOnlyUsernameAvailable** [reply]
Ah.  I can't believe I missed that.  Thank you!
